doc/CI_Hardware.md

```diff
@@ -0,0 +1,221 @@
+# Tock Hardware-in-the-Loop CI
+In addition to software tests, Tock allows platform maintainers to deploy CI runners for hardware platforms, to ensure continuous on-metal performance correctness. The hardware CI is federated, and maintainers for each hardware platform are responsible for ensuring that their hardware CI system is set up, accessible, and runs correctly.
+
+The Tock core team maintains the following hardware CI instances:
+- Nrf52840dk	UC San Diego; contact: Pat Pannuto <ppannuto@ucsd.edu>, Anthony Quiroga <aquiroga@ucsd.edu> 
+- Hail (WIP)	UC San Diego; contact: Pat Pannuto <ppannuto@ucsd.edu>, Anthony Quiroga <aquiroga@usd.edu>
```

Comment:
Can we turn these into links to the boards in the boards folder, i.e. something like `[Nrf52840dk](../boards/nordic/nrf52840dk)` should work